### PR TITLE
Add config.log.format validation

### PR DESCRIPTION
**Pull request by krhubert**
close https://github.com/mesg-foundation/engine/issues/1387

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ type Config struct {
 	}
 
 	Log struct {
-		Format      string `validate:"required"`
+		Format      string `validate:"required,oneof=json text"`
 		ForceColors bool
 		Level       string `validate:"required"`
```
Comment (antho1404) [inline config/config.go]: Should we do that for the log level too?

Comment (NicolasMahe) [reply config/config.go]: No because logrus already have a function that check it and it is used in the `config.validate` function.

**Comment by NicolasMahe** [comment]
@krhubert Don't forget to remove the now useless test lines L163-R165:
https://github.com/mesg-foundation/engine/blob/43514a9e1c01b21c77c98deb74c12c5cfb3ef2ab/config/config.go#L163-L165

File: config/config.go (hunk 1 above)
Comment (antho1404) [reply config/config.go]: So we could remove the `logrus.ParseLevel` test and just keep the test on the config like that we have consistent errors.

Comment (krhubert) [reply config/config.go]: So I will change the error format then to keep it consistent, ok for you? :)

Comment (krhubert) [reply config/config.go]: Or maybe not, the validation errors are more just not so good. 

`Key: 'Config.Log.Format' Error:Field validation for 'Format' failed on the 'oneof' tag`

I would rather change the validation error format to make it more human friendly.

Comment (antho1404) [reply config/config.go]: hum ok, just keep it like it is it's fine

**Pull request merged**